File: Android/FoodRescue/app/src/main/java/edu/ewu/team1/foodrescue/FoodEvent.java
package edu.ewu.team1.foodrescue;

import android.support.annotation.NonNull;

public class FoodEvent implements Comparable<FoodEvent> {
    String title;
    String body;
    private String lat;
    private String lng;
    long timestamp;

    public FoodEvent(String title, String body, String lat, String lng, long timestamp) {
        this.title = title;
        this.body = body;
        this.lat = lat;
        this.lng = lng;
        this.timestamp = timestamp;
    }

    public FoodEvent(String parseMe) {
        String[] data = parseMe.split(",");
        title = data[0];
        body = data[1];
        lat = data[2];
        lng = data[3];
        timestamp = Long.parseLong(data[4]);
    }

    @Override
    public String toString() {
        return title + "," + body + "," + lat + "," + lng + "," + timestamp + "\n";
    }

    @Override
    public int compareTo(@NonNull FoodEvent o) {
        if (this.timestamp - o.timestamp == 0) return 0;
        if ((this.timestamp - o.timestamp) < 0) return -1;
        return 1;
    }

    public double getLat() {
        return Double.parseDouble(lat);
    }

    public double getLng() {
        return Double.parseDouble(lng);
    }
}

File: Android/FoodRescue/app/src/main/java/edu/ewu/team1/foodrescue/FoodEvent.kt
package edu.ewu.team1.foodrescue

class FoodEvent : Comparable<FoodEvent> {
	internal var title: String
	internal var body: String
	var lat: Double
	var lng: Double
	var duration: Int
	internal var timestamp: Long = 0

	constructor(title: String?, body: String?, data: String?, timestamp: Long) {
		this.title = title ?: ""
		this.body = body ?: ""
		val d = data?.split(",")?.toTypedArray() ?: arrayOf("47.491355", "-117.582798", "15")
		this.lat = d[0].toDouble()
		this.lng = d[1].toDouble()
		this.duration = d[2].toInt()
		this.timestamp = timestamp
	}

	constructor(parseMe: String) {
		val data = parseMe.split(",").toTypedArray()
		title = data[0]
		body = data[1]
		lat = data[2].toDouble()
		lng = data[3].toDouble()
		duration = data[4].toInt()
		timestamp = java.lang.Long.parseLong(data[5])
	}

	override fun toString(): String {
		return "$title,$body,$lat,$lng,$duration,$timestamp\n"
	}

	override fun compareTo(other: FoodEvent): Int {
		if (this.timestamp - other.timestamp == 0L) return 0
		return if (this.timestamp - other.timestamp < 0) -1 else 1
	}
}
